packages/dataviews/src/dataform-controls/relative-date-control.tsx

```diff
@@ -0,0 +1,106 @@
+/**
+ * External dependencies
+ */
+import clsx from 'clsx';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	BaseControl,
+	SelectControl,
+	__experimentalNumberControl as NumberControl,
+	__experimentalHStack as HStack,
+} from '@wordpress/components';
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { OPERATOR_IN_THE_PAST, OPERATOR_OVER } from '../constants';
+
+interface RelativeDateControlProps {
+	id: string;
+	value: { value?: string | number; unit?: string };
+	onChange: ( value: any ) => void;
+	label: string;
+	hideLabelFromVision?: boolean;
+	options: { value: string; label: string }[];
+	className?: string;
+}
+
+export const TIME_UNITS_OPTIONS = {
+	[ OPERATOR_IN_THE_PAST ]: [
+		{ value: 'days', label: __( 'Days' ) },
+		{ value: 'weeks', label: __( 'Weeks' ) },
+		{ value: 'months', label: __( 'Months' ) },
+		{ value: 'years', label: __( 'Years' ) },
+	],
+	[ OPERATOR_OVER ]: [
+		{ value: 'days', label: __( 'Days ago' ) },
+		{ value: 'weeks', label: __( 'Weeks ago' ) },
+		{ value: 'months', label: __( 'Months ago' ) },
+		{ value: 'years', label: __( 'Years ago' ) },
+	],
+};
+
+export default function RelativeDateControl( {
+	id,
+	value,
+	onChange,
+	label,
+	hideLabelFromVision,
+	options,
+	className,
+}: RelativeDateControlProps ) {
+	const { value: relValue = '', unit = options[ 0 ].value } = value;
+
+	const onChangeValue = useCallback(
+		( newValue: string | undefined ) =>
+			onChange( {
+				[ id ]: { value: Number( newValue ), unit },
+			} ),
+		[ id, onChange, unit ]
+	);
+
+	const onChangeUnit = useCallback(
+		( newUnit: string | undefined ) =>
+			onChange( {
+				[ id ]: { value: relValue, unit: newUnit },
+			} ),
+		[ id, onChange, relValue ]
+	);
+
+	return (
+		<BaseControl
+			id={ id }
+			__nextHasNoMarginBottom
+			className={ clsx( className, 'dataviews-controls__relative-date' ) }
+			label={ label }
+			hideLabelFromVision={ hideLabelFromVision }
+		>
+			<HStack spacing={ 2.5 }>
+				<NumberControl
+					__next40pxDefaultSize
+					className={ `${ className }-number` }
```

Comment:
Can this and the selectcontrol's class below still be `dataviews-controls__relative-date-number` and `*-unit`? There's also some CSS rules we need to update (check for `.dataviews-controls__datetime-number` and `.dataviews-controls__datetime-unit`).

Additionally, I've noticed `over` and `in the past` render slightly different widths for the number control:

https://github.com/user-attachments/assets/a318de0b-5383-4bd5-9489-613fbc1b0c0c